lib/fastlyctl/fetcher.rb

```diff
@@ -54,25 +64,41 @@ def self.api_request(method, path, options={})
         end
       else
         case response.response_code
-          when 400
-            error = "400: Bad API request--got bad request response."
-          when 403
-            error = "403: Access Denied by API. Run login command to authenticate."
-          when 404
-            error = "404: Service does not exist or bad path requested."
-          when 503
-            error = "503: API is offline."
-          else
-            error = "API responded with status #{response.response_code}."
+        when 400
+          error = "400: Bad API request--something was wrong with the request made by FastlyCTL."
+        when 403
+          error = "403: Access Denied by API. Run login command to authenticate."
+        when 404
+          error = "404: Service does not exist or bad path requested."
+        when 503
+          error = "503: Error from Fastly API--see details below."
+        when 0
+          error = "0: Network connection error occurred."
+        else
+          error = "API responded with status #{response.response_code}."
         end
 
         error += " Method: #{method.to_s.upcase}, Path: #{path}\n"
-        error += "Message from API: #{response.response_body}"
+
+        if (options[:use_vnd]) 
+          begin
+            error_resp = JSON.parse(response.response_body)
+          rescue JSON::ParserError
+            error_resp = {"errors" => [{"title" => "Error parsing response JSON","details" => "No further information available. Please file a github issue at https://github.com/fastly/fastlyctl"}]}
```

Comment:
Does this output the JSON in question?

Reply:
In this particular code path there is no JSON. This line is for a JSON parse error.